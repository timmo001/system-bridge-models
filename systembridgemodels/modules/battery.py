"""Battery."""
from __future__ import annotations

from dataclasses import dataclass


@dataclass
class Battery:
    """Battery."""

    is_charging: bool | None = None
    percentage: float | None = None
