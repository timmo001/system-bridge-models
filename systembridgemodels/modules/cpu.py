"""CPU."""
from __future__ import annotations

from dataclasses import dataclass


@dataclass
class CPUFrequency:
    """CPU Frequency."""

    current: float | None = None
    min: float | None = None
    max: float | None = None


@dataclass
class CPUStats:
    """CPU Stats."""

    ctx_switches: int | None = None
    interrupts: int | None = None
    soft_interrupts: int | None = None
    syscalls: int | None = None


@dataclass
class CPUTimes:
    """CPU Times."""

    user: float | None = None
    system: float | None = None
    idle: float | None = None
    interrupt: float | None = None
    dpc: float | None = None


@dataclass
class CPU:
    """CPU."""

    count: int | None = None
    frequency: CPUFrequency | None = None
    frequency_per_cpu: list[CPUFrequency] | None = None
    load_average: float | None = None
    power_package: float | None = None
    power_per_cpu: list[float] | None = None
    stats: CPUStats | None = None
    temperature: float | None = None
    times: CPUTimes | None = None
    times_per_cpu: list[CPUTimes] | None = None
    times_percent: CPUTimes | None = None
    times_percent_per_cpu: list[CPUTimes] | None = None
    usage: float | None = None
    usage_per_cpu: list[float] | None = None
    voltage: float | None = None
