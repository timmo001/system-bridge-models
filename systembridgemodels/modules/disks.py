"""Disks."""
from __future__ import annotations

from dataclasses import dataclass


@dataclass
class DiskIOCounters:
    """Disk IO Counters."""

    read_count: int
    write_count: int
    read_bytes: int
    write_bytes: int
    read_time: int
    write_time: int


@dataclass
class DiskUsage:
    """Disk Usage."""

    total: int
    used: int
    free: int
    percent: float


@dataclass
class DiskPartition:
    """Disk Partition."""

    device: str
    mount_point: str
    filesystem_type: str
    options: str
    max_file_size: int
    max_path_length: int
    usage: DiskUsage | None = None


@dataclass
class Disk:
    """Disk."""

    name: str
    partitions: list[DiskPartition]
    io_counters: DiskIOCounters | None = None


@dataclass
class Disks:
    """Disks."""

    devices: list[Disk]
    io_counters: DiskIOCounters | None = None
