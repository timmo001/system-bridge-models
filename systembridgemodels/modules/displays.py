"""Displays."""
from __future__ import annotations

from dataclasses import dataclass


@dataclass
class Display:
    """Display."""

    name: str
    resolution_horizontal: int
    resolution_vertical: int
    x: int
    y: int
    width: int | None = None
    height: int | None = None
    is_primary: bool | None = None
    pixel_clock: float | None = None
    refresh_rate: float | None = None


type Displays = list[Display]
