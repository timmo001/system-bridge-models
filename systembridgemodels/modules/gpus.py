"""GPUs."""
from __future__ import annotations

from dataclasses import dataclass


@dataclass
class GPU:
    """GPU."""

    name: str
    core_clock: float | None = None
    core_load: float | None = None
    fan_speed: float | None = None
    memory_clock: float | None = None
    memory_load: float | None = None
    memory_free: float | None = None
    memory_used: float | None = None
    memory_total: float | None = None
    power_usage: float | None = None
    temperature: float | None = None


type GPUs = list[GPU]
