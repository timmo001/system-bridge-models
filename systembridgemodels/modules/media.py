"""Media."""
from __future__ import annotations

from dataclasses import dataclass


@dataclass
class Media:
    """Media Info."""

    album_artist: str | None = None
    album_title: str | None = None
    artist: str | None = None
    duration: float | None = None
    is_fast_forward_enabled: bool | None = None
    is_next_enabled: bool | None = None
    is_pause_enabled: bool | None = None
    is_play_enabled: bool | None = None
    is_previous_enabled: bool | None = None
    is_rewind_enabled: bool | None = None
    is_stop_enabled: bool | None = None
    playback_rate: float | None = None
    position: float | None = None
    repeat: str | None = None
    shuffle: bool | None = None
    status: str | None = None
    subtitle: str | None = None
    thumbnail: str | None = None
    title: str | None = None
    track_number: int | None = None
    type: str | None = None
    updated_at: float | None = None
