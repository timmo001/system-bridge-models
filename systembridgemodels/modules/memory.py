"""Memory."""
from __future__ import annotations

from dataclasses import dataclass


@dataclass
class MemorySwap:
    """Memory Swap."""

    total: int | None = None
    used: int | None = None
    free: float | None = None
    percent: float | None = None
    sin: int | None = None
    sout: int | None = None


@dataclass
class MemoryVirtual:
    """Memory Virtual."""

    total: int | None = None
    available: int | None = None
    percent: float | None = None
    used: int | None = None
    free: int | None = None
    active: int | None = None
    inactive: int | None = None
    buffers: int | None = None
    cached: int | None = None
    wired: int | None = None
    shared: int | None = None


@dataclass
class Memory:
    """Memory."""

    swap: MemorySwap
    virtual: MemoryVirtual
