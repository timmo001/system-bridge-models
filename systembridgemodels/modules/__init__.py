"""Modules."""
from __future__ import annotations

from dataclasses import dataclass
from enum import Enum

from .battery import Battery
from .cpu import CPU
from .disks import Disks
from .displays import Displays
from .gpus import GPUs
from .media import Media
from .memory import Memory
from .networks import Networks
from .processes import Processes
from .sensors import Sensors
from .system import System


class DataEnum(Enum):
    """Data Enum."""

    BATTERY = "battery"
    CPU = "cpu"
    DISKS = "disks"
    DISPLAYS = "displays"
    GPUS = "gpus"
    MEDIA = "media"
    MEMORY = "memory"
    NETWORKS = "networks"
    PROCESSES = "processes"
    SENSORS = "sensor"
    SYSTEM = "system"


@dataclass
class GetData:
    """Get Data."""

    modules: list[str]


@dataclass
class RegisterDataListener(GetData):
    """Register Data Listener."""


@dataclass
class ModulesData:
    """Data."""

    battery: Battery | None = None
    cpu: CPU | None = None
    disks: Disks | None = None
    displays: Displays | None = None
    gpus: GPUs | None = None
    media: Media | None = None
    memory: Memory | None = None
    networks: Networks | None = None
    processes: Processes | None = None
    sensors: Sensors | None = None
    system: System | None = None
