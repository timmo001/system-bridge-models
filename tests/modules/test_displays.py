"""Test the displays module model."""

from syrupy.assertion import SnapshotAssertion

from systembridgemodels.fixtures.modules.displays import FIXTURE_DISPLAYS
from systembridgemodels.modules.displays import Display


def test_displays(snapshot: SnapshotAssertion):
    """Test the displays model."""
    displays = FIXTURE_DISPLAYS
    assert isinstance(displays, list)
    assert isinstance(displays[0], Display)
    assert displays == snapshot
