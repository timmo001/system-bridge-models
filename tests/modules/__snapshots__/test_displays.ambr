# serializer version: 1
# name: test_displays
  list([
    Display(id='abc123', name='name', resolution_horizontal=1920, resolution_vertical=1080, x=1920, y=1080, width=1920, height=1080, is_primary=True, pixel_clock=60.0, refresh_rate=60.0),
  ])
# ---
